server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -233,6 +256,29 @@ private RemoteRestoreResult executeRestore(
         return RemoteRestoreResult.build(restoreUUID, restoreInfo, allocationService.reroute(updatedState, "restored from remote store"));
     }
 
+    private ClusterState restoreClusterMetadata(ClusterState restoredClusterState, Metadata clusterMetadata) {
+        Metadata.Builder mdBuilder = new Metadata.Builder();
+        if (clusterMetadata.persistentSettings() != null) {
+            Settings settings = clusterMetadata.persistentSettings();
+            clusterService.getClusterSettings().validateUpdate(settings);
+            mdBuilder.persistentSettings(settings);
+        }
+        if (clusterMetadata.templates() != null) {
+            for (final IndexTemplateMetadata cursor : clusterMetadata.templates().values()) {
+                mdBuilder.put(cursor);
+            }
+        }
+        if (clusterMetadata.customs() != null) {
+            for (final Map.Entry<String, Metadata.Custom> cursor : clusterMetadata.customs().entrySet()) {
+                if (RepositoriesMetadata.TYPE.equals(cursor.getKey()) == false) {
+                    // TODO: we should be restoring repositories as well after validating no conflict with remote repo
```

Comment:
We should just restore repository metadata, excluding the restricted settings, which would guarantee immutability

Reply:
@Bukhtawar did not understand this one fully. why do we need to exclude the restricted settings?